### PR TITLE
typescript-fetch: Fix compatibility with noImplicitAny TS rule

### PR DESCRIPTION
**Pull request by dngconsulting**
## Why this PR 
runtime.ts file generated by typescript-fetch template doesn't meet the needs of Typescript rule "noImplicitAny" that requires that any function should return an object with its type.
With the current version, the user gets a compile error and it's not possible to fix this error by removing the TS rule. In CRA (Create React Apps and react-script) this rule is kind of "hardcoded" in tsconfig.json. The rules are overridden each time a compilation is raised.

## How it solve the issue ?
Just by adding the correct return types. It doesn't break anything in the current code, it just ensures that correct types will be returned.

### CODE REVIEW

**Comment by macjohnny** [comment]
please run `bin/typescript-fetch-petstore-all.sh` and commit the changed samples

**Comment by dngconsulting** [comment]
I can't, this is a shell script and my laptop is Windows 10  => 

```# START SCRIPT: ./typescript-fetch-petstore.sh
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.065 s
[INFO] Finished at: 2019-12-05T17:39:13+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (C:\Users\Sami\projets\openapi-generator\bin). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
Error: Unable to access jarfile ./modules/openapi-generator-cli/target/openapi-generator-cli.jar
cp: cannot stat 'CI/samples.ci/client/petstore/typescript-fetch/builds/default/pom.xml': No such file or directory
cp: cannot stat 'CI/samples.ci/client/petstore/typescript-fetch/tests/default/pom.xml': No such file or directory
```

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 1]
```diff
@@ -275,31 +275,31 @@ export interface ResponseTransformer<T> {
 export class JSONApiResponse<T> {
     constructor(public raw: Response, private transformer: ResponseTransformer<T> = (jsonValue: any) => jsonValue) {}
 
-    async value() {
+    async value() : Promise<any> {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]: how about
```suggestion
    async value(): Promise<T> {
```
?

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 2]
```diff
@@ -275,31 +275,31 @@ export interface ResponseTransformer<T> {
...
         return this.transformer(await this.raw.json());
     }
 }
 
 export class VoidApiResponse {
     constructor(public raw: Response) {}
 
-    async value() {
+    async value() : Promise<any> {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]:
```suggestion
    async value(): Promise<void> {
```

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 3]
```diff
@@ -275,31 +275,31 @@ export interface ResponseTransformer<T> {
...
         return undefined;
     }
 }
 
 export class BlobApiResponse {
     constructor(public raw: Response) {}
 
-    async value() {
+    async value() : Promise<Blob>{
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]:
```suggestion
    async value(): Promise<Blob>{
```

File: modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache [hunk 4]
```diff
@@ -275,31 +275,31 @@ export interface ResponseTransformer<T> {
...
         return await this.raw.blob();
     };
 }
 
 export class TextApiResponse {
     constructor(public raw: Response) {}
 
-    async value() {
+    async value() : Promise<string>{
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache]:
```suggestion
    async value(): Promise<string>{
```

**Comment by macjohnny** [comment]
please re-generate the samples with `bin/typescript-fetch-petstore-all.sh`

**Comment by dngconsulting** [comment]
For some reason `bin/typescript-fetch-petstore-all.sh` didn't regenerate all ts files (probably because the diff is the same after trimming). I changed the files manually to remove the extra space so please have a look.

**Pull request merged**